_rules/form-field-visual-label-is-descriptive-1cd4da.md

```diff
@@ -0,0 +1,210 @@
+---
+id: 1cd4da
+name: Form field visual label is descriptive
+rule_type: atomic
+description: |
+  This rule checks that visual labels describe the purpose of form field elements.
+accessibility_requirements:
+  wcag21:2.4.6: # Headings and labels (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+authors:
+  - Anne Thyme Nørregaard
+  - Dagfinn Rømen
+  - Geir Sindre Fossøy
+  - Jean-Yves Moyen
+---
+
+## Applicability
+
+This rule applies to any HTML or SVG element that:
+
+- has one of the following [semantic roles][semantic role]: `checkbox`, `combobox` (`select` elements), `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider`, `spinbutton`, `switch` and `textbox`; and
+- is [visible][] or [included in the accessibility tree][]; and
+- is the [labeled control](https://html.spec.whatwg.org/multipage/forms.html#labeled-control) of a [visible][] `label` element.
```

Comment:
According to WCAG, the visible label does not have to be a label element (see three notes below). Either update the applicability, or remove the note (I am in favour of the former). However, this will mean that a lot of the inapplicable test cases must be (re)moved.

Reply:
The rule is explicitly targeting `label` elements, not "label" in the general WCAG sense, hence the need for the note.

My guess is that targeting `label` elements can easily be automated, thus making it a semi-automatic rule. On the other hand, targeting "label" in the WCAG sense of the term cannot be automated and the rule would be manual.

The targeting specifically on `label` elements, as well as the note about WCAG "label" was already here in the [current version of the rule](https://act-rules.github.io/rules/cc0f0a)

The result is, of course, only a partial check for SC 2.4.6.
We might want to add a new manual rule to complete the check (on labels, at least), but I think this should be a separate PR since this would be a complete new rule.

Reply:
I think you could add the brief sentence "This is a partial check for SC 2.4.6 that can be automated" (or such) to the beginning of the note, to make this rationale clearer.

Reply:
Check also https://github.com/act-rules/act-rules.github.io/pull/905#issuecomment-534526764